### PR TITLE
Fix building on Windows 7

### PR DESCRIPTION
**Pull request by AnErrupTion**
Source : https://github.com/NuGet/NuGetGallery/issues/8176#issuecomment-683923724

This enables TLS 1.2 in the registry if a certain update is installed.

### CODE REVIEW

File: install-VS2019.bat [hunk 1]
```diff
@@ -1,28 +1,51 @@
...
-
-echo Building Builder.sln
-"%MSBuild%" Builder.sln /nologo /maxcpucount /nodeReuse:false /verbosity:minimal /t:Restore;Build
-
-start "Cosmos Builder" "source\Cosmos.Build.Builder\bin\Debug\Cosmos.Build.Builder.exe" "-VSPATH=%InstallDir%" %*
+@echo off
+setlocal
+for /f "tokens=4-5 delims=. " %%i in ('ver') do set VERSION=%%i.%%j
+
+if "%version%" == "6.1" goto seven
```
Comment (quajak) [inline install-VS2019.bat:5]: Should this not check for version != 6.1 and then goto start?

File: install-VS2019.bat [hunk 2]
```diff
@@ -1,28 +1,51 @@
...
+
+:seven
+echo Detected Windows 7! Checking for update KB3140245...
+
+for /f "delims=" %%i in ("wmic qfe | find /c ""KB3140245""") do set update=%%i
+if "%update%" == "0" goto update404
```
Comment (quajak) [inline install-VS2019.bat:11]: Can you please use an if block for what to do and not a goto

**Review by quajak — CHANGES_REQUESTED** [review]
Thanks for the PR.

File: install-VS2019.bat (hunk 1 above)
Comment (AnErrupTion) [reply install-VS2019.bat:5]: it checks if the NT version is 6.1, if yes then it goes to :seven and checks for a specific update, etc

Comment (quajak) [reply install-VS2019.bat:5]: My understanding of batch might be wrong, but doesnt the code end up executing the seven block even if the NT version is not 6.1?

Comment (AnErrupTion) [reply install-VS2019.bat:5]: ah that's true lol, didn't see that one

**Review by quajak — APPROVED** [review]

**Pull request merged**